MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -0,0 +1,145 @@
+        let currentRoadName = wayNameView.text!
+        XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+        XCTAssertFalse(wayNameView.isHidden, "WayNameView should be visible.")
+    }
+    
+    // Brief: navigationViewController(_:roadNameAt:) delegate method is implemented,
+    //        with a blank road name (empty string) provided and wayNameView label is hidden.
+    func testNavigationViewControllerDelegateRoadNameAtLocationEmptyString() {
+        
+        let navigationViewController = dependencies.navigationViewController
+        let routeController = navigationViewController.routeController!
+        
+        // Identify a location to set the custom road name.
+        let turkStreetLocation = dependencies.poi[1]
+        let roadName = ""
+        customRoadName[turkStreetLocation.coordinate] = roadName
+        
+        routeController.locationManager(routeController.locationManager, didUpdateLocations: [turkStreetLocation])
+        
+        let wayNameView = (navigationViewController.mapViewController?.navigationView.wayNameView)!
+        let currentRoadName = wayNameView.text!
+        XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+        XCTAssertTrue(wayNameView.isHidden, "WayNameView should be hidden.")
+    }
+    
+    func testNavigationViewControllerDelegateRoadNameAtLocationUmimplemented() {
+        
+        let navigationViewController = dependencies.navigationViewController
+        
+        // We break the communication between CLLocation and MBRouteController
+        // Intent: Prevent the routecontroller from being fed real location updates
+        navigationViewController.routeController.locationManager.delegate = nil
+        
+        UIApplication.shared.delegate!.window!!.addSubview(navigationViewController.view)
+        
+        let routeController = navigationViewController.routeController!
+        
+        // Identify a location without a custom road name.
+        let fultonStreetLocation = dependencies.poi[2]
+        
+        navigationViewController.mapViewController!.labelRoadNameCompletionHandler = { (defaultRaodNameAssigned) in
+            XCTAssertTrue(defaultRaodNameAssigned, "Unfortunstely label road name was not successfully set")
+        }
+        
+        routeController.locationManager(routeController.locationManager, didUpdateLocations: [fultonStreetLocation])
+    }
+}
+
+extension NavigationViewControllerTests: NavigationViewControllerDelegate {
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, roadNameAt location: CLLocation) -> String? {
+        return customRoadName[location.coordinate] ?? nil
+    }
+}
+
+extension CLLocationCoordinate2D: Hashable {
+    // Hash value property multiplied by a prime constant.
+    public var hashValue: Int {
```

Comment:
Is this used?

Reply:
Yep, implicitly for our key-value pairs here:
https://github.com/mapbox/mapbox-navigation-ios/pull/1338/files/ea482a1284403cfbb4e4260eee3c62b692bd7162#diff-841adf0b96176970740abfa00fe996b2R7

Reply:
@bsudekum Yes, it will. Property 'hashValue' must be declared public because it matches a requirement in public protocol 'Hashable'.